bpf/aya-bpf/src/helpers.rs

```diff
@@ -836,3 +840,29 @@ pub unsafe fn bpf_printk_impl<const FMT_LEN: usize, const NUM_ARGS: usize>(
         _ => gen::bpf_trace_vprintk(fmt_ptr, fmt_size, args.as_ptr() as _, (NUM_ARGS * 8) as _),
     }
 }
+
+/// Compares the given byte `s1` with a [`&CStr`](core::ffi::CStr) `s2`.
+///
+/// # Examples
+///
+/// ```no_run
+/// # use aya_bpf::helpers::bpf_strncmp;
+/// # let data = b"something";
+/// if bpf_strncmp(data, c"foo") == core::cmp::Ordering::Equal {
+///     // The comparison succeeded.    
+/// }
+/// ```
+#[inline]
+pub fn bpf_strncmp(s1: &[u8], s2: &CStr) -> Ordering {
```

Comment:
I don't understand the API why one is a slice and the other is &CStr?

Reply:
Well, the most common way of using `bpf_strncmp` I've seen (and I'm doing in my projects :stuck_out_tongue: ) is comparing a string we got from `bpf_probe_read_user_str_bytes` (which returns a byte slice) with some literal written in the program (where the CStr literal is very handy).

But yeah, I get your point about consistency. Since our whole `bpf_probe_read_*_bytes` API returns byte slices, I see two solutions:

* We just use byte slices. And tell people to use byte literals - I'm mostly convinced to that so far.
* We make everything return CStr - but nah, we would need to care about null termination checks and so.

Reply:
The documentation says the left operand doesn't need to be null-terminated:

https://github.com/torvalds/linux/blob/f89a687aaeae54fb59e8e8aa619bd448a9c235a2/include/uapi/linux/bpf.h#L5389-L5393

```
 * long bpf_strncmp(const char *s1, u32 s1_sz, const char *s2)
 *	Description
 *		Do strncmp() between **s1** and **s2**. **s1** doesn't need
 *		to be null-terminated and **s1_sz** is the maximum storage
 *		size of **s1**. **s2** must be a read-only string.
 *	Return
 *		An integer less than, equal to, or greater than zero
 *		if the first **s1_sz** bytes of **s1** is found to be
 *		less than, to match, or be greater than **s2**.
```